src/main/java/top/seraphjack/simplelogin/server/ServerSideEventHandler.java

```diff
@@ -28,6 +31,19 @@ public static void playerJoin(PlayerEvent.PlayerLoggedInEvent event) {
         });
     }
 
+    @SubscribeEvent
+    public static synchronized void playerLeave(PlayerEvent.PlayerLoggedOutEvent event) {
+        final String username = event.player.getName();
+        final Position pos = new Position(event.player.posX, event.player.posY, event.player.posZ);
+        SLStorage.instance().storageProvider.setLastPosition(username, pos);
+        try {
+            BlockPos spawnPoint = event.player.world.getSpawnPoint();
+            event.player.setPosition(spawnPoint.getX(), spawnPoint.getY(), spawnPoint.getZ());
+        } catch (Throwable ex) {
+            SimpleLogin.logger.error("Fail to process logout.", ex);
```

Comment:
`SimpleLogin.logger.error("Fail to set player position to spawn point when logging out.", ex);`